src/Raven.Server/Json/BlittableNumber.cs

```diff
@@ -24,6 +24,14 @@ public static NumberParseResult Parse(object value, out double doubleResult, out
                 return NumberParseResult.Double;
             }
 
+            if (value is float)
+            {
+                longResult = long.MinValue;
+                doubleResult = (double)(decimal)(float)value;
```

Comment:
The conversion from float to double adds garbage to the converted number.
https://stackoverflow.com/questions/5986980/convert-float-to-double-loses-precision-but-not-via-tostring